src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneCreditCardCheckMethodSender.php

```diff
@@ -0,0 +1,85 @@
+use SprykerEco\Zed\Payone\Business\Api\Response\Mapper\CreditCardCheckResponseMapper;
+use SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface;
+use SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader;
+
+class PayoneCreditCardCheckMethodSender implements PayoneCreditCardCheckMethodSenderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface
+     */
+    protected $executionAdapter;
+
+    /**
+     * @var \Generated\Shared\Transfer\PayoneStandardParameterTransfer
+     */
+    protected $standardParameter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMethodMapperInterface[]
+     */
+    protected $registeredMethodMappers;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface
+     */
+    protected $standartParameterMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader
+     */
+    protected $paymentMapperReader;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface $executionAdapter
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneStandardParameterTransfer $standardParameter,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PaymentMapperReader $paymentMapperReader
+    ) {
+        $this->executionAdapter = $executionAdapter;
+        $this->standardParameter = $standardParameter;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->paymentMapperReader = $paymentMapperReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneCreditCardTransfer $creditCardData
+     *
+     * @return \Generated\Shared\Transfer\CreditCardCheckResponseTransfer
+     */
+    public function creditCardCheck(PayoneCreditCardTransfer $creditCardData): CreditCardCheckResponseTransfer
+    {
+        /** @var \SprykerEco\Zed\Payone\Business\Payment\MethodMapper\CreditCardPseudo $paymentMethodMapper */
+        $paymentMethodMapper = $this->paymentMapperReader->getRegisteredPaymentMethodMapper($creditCardData->getPayment()->getPaymentMethod());
+        $requestContainer = $paymentMethodMapper->mapCreditCardCheck($creditCardData);
+
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+        $responseContainer = new CreditCardCheckResponseContainer($rawResponse);
+
+        $responseMapper = new CreditCardCheckResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
++